### PR TITLE
Deprecate monotonicity_helper

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Related to the discusssion in #23037

#### Brief description of what is fixed or changed
This is a typical example of what I believe should be "private". Currently, this function is in the documentation leading to that there is yet another function listed that a user shouldn't really need to access directly (and the list of functions that a user should access is quite long as it is).

Here, this function is well abstracted and well documented.

#### Other comments
This clearly only has a minor impact on the whole issue (and the user experience), but there are quite a few functions and methods like this in the code base. Things that probably should have been private to start with.

#### Release Notes

* calculus
    * BREAKING CHANGE: `monotonicity_helper` is deprecated.

### CODE REVIEW

File: sympy/calculus/singularities.py [hunk 1]
```diff
@@ -109,7 +110,47 @@ def singularities(expression, symbol, domain=None):
...
 
 
+@deprecated(
+    "monotonicity_helper is considered a private function, use the direct"
+    " function for checking instead", deprecated_since_version="1.12",
+    active_deprecations_target="monotonicity-helper"
+)
 def monotonicity_helper(expression, predicate, interval=S.Reals, symbol=None):
+    """
+    .. deprecated:: 1.12
```
Comment (oscarbenjamin) [inline sympy/calculus/singularities.py:120]: This should probably be 1.13 unless you plan to backport this PR to the 1.12 branch that I just created.

Comment (oscargus) [reply sympy/calculus/singularities.py:120]: Ahh, I checked and was a bit confused about the release status.

Comment (oscarbenjamin) [reply sympy/calculus/singularities.py]: I haven't quite finished setting everything up on the branches: #24776

Comment (oscargus) [reply sympy/calculus/singularities.py]: Maybe we should try to have some automatic backporting? So if a PR is marked 1.12 it will be backported upon merging? I think https://github.com/meeseeksbox is a rather good option. 

Since I just pushed a number of doc-fixes it can make even more sense for those, rather than having to (more or less) manually backport stuff. Also convenient for a bugfix releases.

Comment (oscarbenjamin) [reply sympy/calculus/singularities.py]: It would be good to make it easier but it's quite unusual that anything really needs to be backported.

It's possible that making it easier would mostly just mean that things get backported unnecessarily.

Comment (oscargus) [reply sympy/calculus/singularities.py]: Could be. But depends on how one defines unnecessarily.

I guess it is primarily used for things one would like to get in either after the fork (bug fixes from RCs, documentation improvements) or if one tend to release regression fixes  (which we rarely do).

Right now there are not that many PRs marked 1.12 though.

Comment (oscargus) [reply sympy/calculus/singularities.py]: #24800 is a typical thing that could be backported at this stage if it was reasonably easy. Minor thing, but makes sense to get in. (And I realize that "it will never break anything"-stuff is not always good to get in with short notice...)

Comment (oscarbenjamin) [reply sympy/calculus/singularities.py]: SymPy's releases are basically just a snapshot of the master branch at some more or less arbitrarily chosen point in time. If there is any value in having a release branch then it is so that we can limit what changes are applied to it. In that sense we should only backport things that are urgent bugs or fixes for regressions that are picked up after the release so that we minimise the chance of introducing any new bugs.

In other words the release branch primarily exists just so that we can assure downstream users that the release matches the release candidate and that any bugfix releases only contain bugfixes.

Comment (oscarbenjamin) [reply sympy/calculus/singularities.py]: > #24800 is a typical thing that could be backported at this stage

I wouldn't classify that one as having any urgency. That's adding a helpful error message in case someone uses `nsolve` incorrectly which is nice to do but not really what I would call a bugfix. If say it was a new bug since the last release that meant that `nsolve` now returns an incorrect result *when used correctly* then I would say that it should be backported.

Comment (oscargus) [reply sympy/calculus/singularities.py]: Yeah, one can discuss what could/should go in... Personally, I'd backport "non-controversial improvements" (for some definition) during the preparation period, but then I am not releasing anything (which is probably good for many other reasons, possibly including this).

**Comment by sylee957** [comment]
I'm not sure whether there is a more strong argument about deprecating it though, maybe because it has name 'helper'?
I'm not sure about whether the original commit got reviewed good at the first place, but at least I'm not sure we can provide any alternative to `predicate=lambda x: x > 1` or such at the point now
